### PR TITLE
chore: updates to readme and code.

### PR DESCRIPTION
**Pull request by lholmquist**
* This adds a section on setting up the kafka operator
* Adds the ability to pass env vars for the cluster location
* Fixes creating a new producer for every message call

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,81 +10,62 @@ This is composed by:
 
 > The example is intended to be deployed and run in an Openshift or [CodeReady containers](https://developers.redhat.com/products/codeready-containers/overview) with [Kafka operator](https://strimzi.io/quickstarts/) installed. (Although it is possible to run on a common kafka installation.)
 
+## Setup Kafka Operator
+
+Installing the Kafka Operator using the Web Console of Openshift is pretty straightforward.
+
+First, go to the *Operator Hub* and search for *Kafak*.  A few choices might show up,  but we want the one that says "Strimzi" which is the community Kafka version.
```
Comment (mhdawson) [inline README.md:17]:
```suggestion
First, go to the *Operator Hub* and search for *Kafka*.  A few choices might show up,  but we want the one that says "Strimzi" which is the community Kafka version.
```

File: README.md [hunk 2]
```diff
@@ -10,81 +10,62 @@ This is composed by:
...
-A namespace `nodejs-kafka` was created in Openshift to deploy the tree services. So the producer's `bootstrap.servers` are 
-configured like this:
+If you used the defaults for the cluster naming, then nothing needs to change, but if you named the cluster something else, then you will need to change the server name in the code:
 
 ```
 new Kaka.Producer({
-  'bootstrap.servers': 'nodejs-kafka-cluster-kafka-bootstrap:9092'
+  'bootstrap.servers': 'my-cluster-kafka-bootstrap:9092'
 })
 ```
```
Comment (mhdawson) [inline README.md:46]: This needs to be updated to reflect the changes you made to add `process.env.KAFKA_BOOTSTRAP_SERVER`.  I think we can also say in the text above something like "you may have to update the server name. Some environments like openshift will automatically inject the server name, if that's not the case for you environment you'll need to update as follows"

**Comment by lholmquist** [comment]
@mhdawson i still need to make the update regarding the env vars,  but I just added a separate setup guide for the Operator.  with Pictures!

**Comment by helio-frota** [comment]
@lholmquist great for the PR 

I've changed the environment variable to `KAFKA_BOOTSTRAP_SERVER="nodejs-kafka-cluster-kafka-bootstrap:9092"`
and the deploy works fine with `./deploy-to-openshift.sh`

I'm getting `producer-backend-1-qvbpx | CrashLoopBackOff` error, but that seems is not related to the PR. 

I'm going to cleanup/restart to see what happen

**Comment by helio-frota** [comment]
@lholmquist I was wrong when I reverted the producer.js code it stops with the `CrashLoopBackOff` error.
Your PR makes sense to avoid creating a producer every time, but something is happening.
Does `CrashLoopBackOff` that error happen in your environment? Probably my environment has issues with kafka operator I don't know.

**Comment by lholmquist** [comment]
I think it only happened when i had the wrong service name for the bootstrap server.  If you just accept the defaults when adding the operator and the kakfa instance(make sure to create this in the same namespace as the application),  it should work without any environment variable updates

**Comment by helio-frota** [comment]
yeah it seems to be something with my kafka operator installation.
that is an opportunity to try out your docs to setup a new kafka operator environment (something may be updated over time also)

**Review by helio-frota — APPROVED** [review]

**Pull request merged**